### PR TITLE
Simplify logging by removing `is_*` logging level checks before logging

### PR DESCRIPTION
**Pull request by ehuelsmann**
# Description

The commits in this PR are of gradually increasing logging complexity, where more performance impact might be expected. The idea is to use these commits to benchmark performance of the change and use the outcomes to decide whether or not a specific commit crosses the acceptable threshold.

Fixes/addresses (If applicable) #89

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
This PR mostly simplifies code.

# Checklist:

- [x] My code follows the style guidelines of this project, please see the [contribution guidelines](https://github.com/jonasbn/perl-workflow/blob/master/CONTRIBUTING.md).
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

You might think, that this is one crazy checklist, but it is just as much written for the maintainer of the involved software :-)

### CODE REVIEW

**Comment by jonasbn** [comment]
Hi @ehuelsmann and @oliwel 

I have done some very very unscientific benchmarking of the performance impact and there is no difference between this branch and the master branch. I did basic measurements on repeated test runs. 

I am sure we could do more and @oliwel could give us even better numbers from his platform, with more complex workflows and higher workload, but based on the current test suite I was not able to measure any difference between the two.

If this logging cleanup demonstrates later to have been a bad idea, I guess we can consider reverting out implementation of evaluate better/nicer logging frameworks.

So am all for an inclusion in the upcoming release.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,8 +41,6 @@ jobs:
         run: |
           cd build
           cpanm --installdeps --local-lib ${GITHUB_WORKSPACE}/local --with-all-features --with-develop --with-suggests .
-      - name: Install
-        run: cpanm Devel::Cover::Report::Coveralls
```
Comment (jonasbn) [inline .github/workflows/ci.yml:45]: Any particular reason why this change is a part of this PR?

**Review by jonasbn — CHANGES_REQUESTED** [review]
I think I have reviewed this before, anyway - just a single comment. No change just for my understanding.

**Comment by ehuelsmann** [comment]
@jonasbn from what I understood from @oliwel before, he expects the impact on his system to be negligible, as bottlenecks are elsewhere (e.g. database throughput). I had interpreted that as a "go" from his use-case.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ehuelsmann) [reply .github/workflows/ci.yml:45]: A few lines up, `Devel::Cover::Report::Coveralls` is already installed. It looked like a merge conflict, so I corrected it. Apparently, I was mistaken, resulting in this superfluous change. Do you want me to revert?

Comment (jonasbn) [reply .github/workflows/ci.yml:45]: Then it is good. I just thought the change stood out
